src/NuGet.Clients/NuGet.PackageManagement.UI/PackageLoadContext.cs

```diff
@@ -46,10 +48,38 @@ internal class PackageLoadContext
             SolutionManager = uiContext.SolutionManagerService;
 
             _installedPackagesTask = PackageCollection.FromProjectsAsync(Projects, CancellationToken.None);
+            _allPackagesTask = NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(() => PackageCollection.FromProjectsIncludeTransitiveAsync(Projects, CancellationToken.None));
         }
 
         public Task<PackageCollection> GetInstalledPackagesAsync() => _installedPackagesTask;
 
+        public async Task<(PackageCollection, PackageCollection)> GetAllPackagesAsync() => await _allPackagesTask;
+
+        // Returns the target frameworks
+        public async Task<List<string>> GetTargetFrameworksAsync()
```

Comment:
I added this here to get target frameworks for the recommender. Would make sense to combine this with GetSupportedFrameworksAsync (below)?

Reply:
I combined them into GetSupportedFrameworksAsync.